pretyping/evarconv.ml

```diff
@@ -1228,6 +1228,7 @@ let solve_unconstrained_impossible_cases env evd =
   Evd.fold_undefined (fun evk ev_info evd' ->
     match ev_info.evar_source with
     | loc,Evar_kinds.ImpossibleCase ->
+       (* Why does this ad-hoc code have to be here instead of cases.ml? *)
```

Comment:
This is a priori a non-local resolution. For compatibility with existing code, it is possible to have an impossible branch given explicitly. The type given to the right-hand side of this impossible branch is arbitrary and not necessarily inferrable by local reasoning. Here is an example:

```
Require Import Vector.
Import VectorNotations.
Check (fun y n (x:t nat (S n)) => match x with a::_ => a | _ => y end) 2.
```
where the type of `y` is an evar tagged `ImpossibleCase`. This evar will be instantiated by `nat` only at the time of application of `2`, i.e. outside the scope of the `match`.